runtime/src/lib.rs

```diff
@@ -101,6 +103,61 @@ impl Runtime {
     }
 }
 
+struct EnvAdapter<'a>{
+    system: &'a mut dyn System,
+}
+
+impl Environment for EnvAdapter<'_> {
+
+    fn elapsed(&self) -> Result<Duration, Error> {
+        unimplemented!();
+    }
+
+    fn read_input(
+        &self,
+        address: usize,
+        buffer: &mut [u8],
+    ) -> Result<(), Error>{
+        unimplemented!();
+    }
+
+    fn write_output(
+        &mut self,
+        address: usize,
+        buffer: &[u8],
+    ) -> Result<(), Error>{
+        unimplemented!();
+    }
+
+    fn log(&mut self, record: &Record<'_>) -> Result<(), Error>{
+        unimplemented!();
+    }
+
+    fn get_variable(&self, name: &str) -> Result<WasmValue, Error>{
+        unimplemented!();
+    }
+
+    fn set_variable(&mut self, name: &str, value: WasmValue) -> Result<(), Error>{
+        unimplemented!();
+    }
+
+
+}
+
+pub struct WasmProcess{
+    program: WasmProgram,
+}
+
+impl Process for WasmProcess{
+     type Fault=Fault;
+     fn poll(&mut self, system: &mut dyn System) -> Transition<Self::Fault>{
+          let mut env_adapter = EnvAdapter{system: system};
+          let _ = self.program.poll(&mut env_adapter);
```

Comment:
I'm guessing under normal circumstances we'd want to translate polling errors into a `Fault`, wouldn't we?

Reply:
Yes :+1:

Reply:
Right now ```enum Fault``` is an empty enum. Should this be filled with fault cases or is this enum more like a marker? I'm note sure how to return an empty enum in Rust anyhow...

Reply:
It's actually impossible for an empty enum to exist at runtime. 

If you think about it, when you create a variable with an enum type the only possible values you can give that variable are the enum variants. If there are no variants, it's not possible for a variable with that type to exist.

A good example of where you'd like to do this is when you need to return a `Result`, but it's not possible for the function to ever fail. See the docs on [`std::convert::Infallible`](https://doc.rust-lang.org/std/convert/enum.Infallible.html) for more.

That said, this function can definitely fail so we'll want to populate `Fault` with the various error cases. Another option would be to just return `rustmatic_wasm::Error`.

Reply:
Added return of ```rustmatic_wasm::Error``` for now until we get some usage experience. 3b99d91

Reply:
Ok, that was wrong. The fault type is constrained to be ```enum Fault``` in ```add_process``` etc. E0271... Added fields to the enum instead. d01e762